### PR TITLE
test: Move MakeNoLogFileContext to libtest_util, and use it in bench

### PR DESCRIPTION
**Pull request by maflcko**
To avoid verbose code duplication, which may lead to accidental mishaps https://github.com/bitcoin/bitcoin/pull/20998/files#r563624041.

Also fix a nit I found in https://github.com/bitcoin/bitcoin/pull/20946#discussion_r561949731.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Doesn't work, because of globals

**Comment by MarcoFalke** [comment]
Worked around #21005

**Comment by MarcoFalke** [comment]
Rebased

**Comment by MarcoFalke** [comment]
Rebased (3x)

**Comment by fanquake** [comment]
concept ack (1x)

**Comment by MarcoFalke** [comment]
Rebased (4x)

**Comment by MarcoFalke** [comment]
Rebased (5x)

**Comment by practicalswift** [comment]
Concept ACK (2x)

File: src/bench/rpc_blockchain.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@
 namespace {
 
 struct TestBlockAndIndex {
-    TestingSetup test_setup{};
+    const std::unique_ptr<const TestingSetup> testing_setup{MakeNoLogFileContext<const TestingSetup>(CBaseChainParams::MAIN)};
```
Comment (fanquake) [inline src/bench/rpc_blockchain.cpp:18]: This seems kind of unruly.

**Review by fanquake — APPROVED** [review]
ACK fa576b4532814b4bca1936d170cabd01fbc51960

File: src/bench/rpc_blockchain.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/bench/rpc_blockchain.cpp:18]: > unruly

C++ requires this. (You can't use `auto` for non-static members)

**Comment by MarcoFalke** [comment]
@dongcarl Mind to take a look here, as this builds on top of  #20946?

**Comment by dongcarl** [comment]
Light Code-Review ACK fa576b4532814b4bca1936d170cabd01fbc51960

Looks reasonable. Always good to coalesce these init codepaths.

**Comment by fanquake** [comment]
Failed merged (1x)

This need fixing up post #19203 & #19288:
```bash
test/fuzz/socks5.cpp:22:66: error: unknown type name 'BasicTestingSetup'
    static const auto testing_setup = MakeNoLogFileContext<const BasicTestingSetup>();
                                                                 ^
1 error generated.
gmake[2]: *** [Makefile:13693: test/fuzz/fuzz-socks5.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
test/fuzz/torcontrol.cpp:38:39: error: use of undeclared identifier 'MakeNoLogFileContext'
    static const auto testing_setup = MakeNoLogFileContext<>();
                                      ^
1 error generated.
```

```diff
diff --git a/src/test/fuzz/socks5.cpp b/src/test/fuzz/socks5.cpp
index 123ee042e..72c3e8f71 100644
--- a/src/test/fuzz/socks5.cpp
+++ b/src/test/fuzz/socks5.cpp
@@ -6,6 +6,7 @@
 #include <test/fuzz/FuzzedDataProvider.h>
 #include <test/fuzz/fuzz.h>
 #include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
 
 #include <cstdint>
 #include <string>
diff --git a/src/test/fuzz/torcontrol.cpp b/src/test/fuzz/torcontrol.cpp
index b7a42ea7f..a8e69dee3 100644
--- a/src/test/fuzz/torcontrol.cpp
+++ b/src/test/fuzz/torcontrol.cpp
@@ -5,6 +5,7 @@
 #include <test/fuzz/FuzzedDataProvider.h>
 #include <test/fuzz/fuzz.h>
 #include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
 #include <torcontrol.h>
 
 #include <cstdint>
```

**Comment by MarcoFalke** [comment]
Not going to invalidate the precious reviews over an unrelated issue. Fixed in #21358, which can go in first.

**Pull request merged**